influx-writer/influx-writer.py

```diff
@@ -58,17 +64,13 @@ def write(self, listen_dicts):
             if t < min_time:
                 min_time = t
 
-        # Quote single quote characters which could be used to mount an injection attack.
-        # Sadly, influxdb does not provide a means to do this in the client library
-        user_name = user_name.replace("'", "\'")
-
         # quering for artist name here, since a field must be included in the query.
         query = """SELECT time, artist_name
-                     FROM listen
-                    WHERE user_name = '%s'
-                      AND time >= %d000000000
+                     FROM "\\"%s\\""
+                    WHERE time >= %d000000000
                       AND time <= %d000000000
                 """ % (user_name, min_time, max_time)
+
```

Comment:
I did it how it was done here: https://github.com/metabrainz/listenbrainz-server/blob/master/listenstore/influx_listenstore.py#L199

I guess we should escape the username there also, I'll fix and send soon.